tests/e2e/matchers/to-have-amp-auto-ads-tag.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Custom matcher for testing <amp-auto-ads> exists on a page.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { createURL } from '@wordpress/e2e-test-utils';
+
+/**
+ * Asserts the URL at the given path contains an <amp-auto-ads> tag.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} path The URL path of the current site to check.
+ * @return {Object} Matcher result.
+ */
+export async function toHaveAMPAutoAdsTag( path ) {
+	const result = {};
+
+	const page = await browser.newPage();
+	await page.goto( createURL( path, 'amp' ) );
```

Comment:
Since the `amp` query param only applies to secondary AMP modes (traditional at least), let's pass that as part of the `path` when calling rather than use this version of the URL unconditionally.

That way we could also test that `expect( pathWithoutAMP ).not.toHaveAMPAutoAdsTag()` when AMP is active and configured to assert that the non-AMP version/URL of the page doesn't render it.

Reply:
@aaemnnosttv I originally had this in place however, the `?amp` was getting escaped so i went this route. I'll add a second optional boolean param that will add `amp` if `true`